### PR TITLE
Adding reply callback to deployFeed and adding using it for canary failures

### PR DESCRIPTION
**Pull request by IanWoodard**
Adding a new callback to the deployFeed which enables returning slack blocks to be sent as a reply to a sent message as a follow up. The first usage of this is sending a reply following a canary failure explaining that the pipeline is paused and tagging the people involved (those who have commits in the failing deploy).

### CODE REVIEW

**Review by mattgauntseo-sentry — CHANGES_REQUESTED** [review]
Will chat about tests in a bit

File: src/brain/gocdSlackFeeds/index.ts [hunk 1]
```diff
@@ -109,6 +120,56 @@ const engineeringFeed = new DeployFeed({
...
+      pipeline.stage.result.toLowerCase() === 'failed'
+    ) {
+      const [base, head] = await getBaseAndHeadCommit(pipeline);
+      if (!head) {
+        return [];
+      }
+      const authors = await getAuthors('getsentry', base, head);
+      // If there are no authors, we can't cc anyone
+      if (authors.length === 0) return [];
+      const uniqueAuthors = authors.filter(
```
[the remark this answers is not in the document]
Comment (IanWoodard) [reply src/brain/gocdSlackFeeds/index.ts]: The one downside to filtering after the getUser call is that we would be making unnecessary calls to getUser (which isn't trivial). But overall, I agree.

**Review by mattgauntseo-sentry — APPROVED** [review]

File: src/brain/gocdSlackFeeds/index.test.ts [hunk 2]
```diff
@@ -43,6 +45,33 @@ describe('gocdSlackFeeds', function () {
   });
 
   it('post and update message to all feeds', async function () {
+    const user = {
+      email: 'test@sentry.io',
+      slackUser: 'U1234',
+      githubUser: 'githubUser',
+    };
+    await db('users').insert(user);
```
[the remark this answers is not in the document]
Comment (IanWoodard) [reply src/brain/gocdSlackFeeds/index.test.ts]: I removed the db usage

**Comment by mattgauntseo-sentry** [comment]
LGTM

**Pull request merged**